letsencrypt/README.rst

```diff
@@ -0,0 +1,164 @@
+
+This module was written to have your Odoo installation request SSL certificates
+from https://letsencrypt.org automatically.
+
+Installation
+============
+
+After installation, this module generates a private key for your account at
+letsencrypt.org automatically in ``$data_dir/letsencrypt/account.key``. If you
+want or need to use your own account key, replace the file.
+
+For certificate requests to work, your site needs to be accessible via plain
+HTTP, see below for configuration examples in case you force your clients to
+the SSL version.
+
+After installation, trigger the cronjob `Update letsencrypt certificates` and
+watch your log for messages.
+
+This addon depends on the ``openssl`` binary and the ``acme_tiny`` and ``IPy``
+python modules.
+
+For installing the OpenSSL binary you can use your distro package manager.
+For Debian and Ubuntu, that would be:
+
+    sudo apt-get install openssl
+
+For installing the ACME-Tiny python module, use the PIP package manager:
+
+    sudo pip install acme-tiny
+
+For installing the IPy python module, use the PIP package manager:
+
+    sudo pip install IPy
+
+
+Configuration
+=============
+
+This addons requests a certificate for the domain named in the configuration
+parameter ``web.base.url`` - if this comes back as ``localhost`` or the like,
+the module doesn't request anything.
+
+If you want your certificate to contain multiple alternative names, just add
+them as configuration parameters ``letsencrypt.altname.N`` with ``N`` starting
+from ``0``. The amount of domains that can be added are subject to `rate
+limiting <https://community.letsencrypt.org/t/rate-limits-for-lets-encrypt/6769>`_.
+
+Note that all those domains must be publicly reachable on port 80 via HTTP, and
+they must have an entry for ``.well-known/acme-challenge`` pointing to your odoo
+instance.
+
+Usage
+=====
+
+The module sets up a cronjob that requests and renews certificates automatically.
+
+After the first run, you'll find a file called ``domain.crt`` in
+``$datadir/letsencrypt``, configure your SSL proxy to use this file as certificate.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/149/8.0
+
+For further information, please visit:
+
+* https://www.odoo.com/forum/help-1
+
+In depth configuration
+======================
+
+This module uses ``openssl`` to generate CSRs suitable to be submitted to
+letsencrypt.org. In order to do this, it copies ``/etc/ssl/openssl.cnf`` to a
+temporary and adapts it according to its needs (currently, that's just adding a
+``[SAN]`` section if necessary). If you want the module to use another configuration
+template, set config parameter ``letsencrypt.openssl.cnf``.
+
+After refreshing the certificate, the module attempts to run the content of
+``letsencrypt.reload_command``, which is by default ``sudo service nginx reload``.
```

Comment:
Having the proxy in the same machine of Odoo is being deprecated in favor of 1-process-per-container solutions such as Docker, which should be the semi-standard from now on, and would work much better with my proposal.

Please don't take it badly, but "I don't want to work more" does not seem very convincing as you may guess. Mostly when the extra work would be nothing on your part because if you already have that setting in your production environments, it's under a noupdate tag, so it would not change; maybe your new instances would require an additional 1-line script, but you will thank me if any day you decide to migrate to Docker (or another proxy).

Please bear in mind we are a community here and everyone has his *how this should be done* book. On such sysadmin-related modules, being as most system-agnostic as possible is very desirable, mostly when the effort is second to none.